tests/test_runner/test_discover_runner.py

```diff
@@ -332,6 +333,40 @@ def test_timings_captured(self):
         self.assertTrue(isinstance(runner.time_keeper, TimeKeeper))
         self.assertIn('test', stderr.getvalue())
 
+    def test_log(self):
+        levels = [logging.NOTSET, logging.DEBUG, logging.INFO, logging.WARNING]
+        verbosities = range(4)
+        msg = 'logging message'
```

Comment:
I created a variable because it had to be checked in the `self.assertIn(msg, stdout.getvalue())` and 
`self.assertNotIn(msg, stdout.getvalue())`. Should I replace those occurences with a hard-coded string?

Reply:
Yeah, in tests it's often okay to duplicate things. (By the way, can you just test equality of strings instead of contains? That would be more precise, and you would just have a single assert line with two different expected values.)

Reply:
Okay. Will make the changes